scanner/decisions/0010-separate-versioning-for-vuln-bundles.md

```diff
@@ -0,0 +1,82 @@
+# 0010 - Separate Versioning Stream for Vulnerability Bundles
+
+- **Author(s):** J. Victor Martins <jvdm@sdf.org>
+- **Created:** [2024-07-22 Mon]
+
+## Status
+
+Status: Accepted
+
+Updates: [#0009](0009-scannerv4-read-manual-vulnerabilities.md)
+
+## Context
+
+A Vulnerability Bundle is an archive containing updates to vulnerability data in Scanner.  They are constantly updated by the Vulnerability Exporter (aka. Exporter), which runs periodically in GitHub's Workflows and uses ClairCore to fetch the security data to convert it to a JSON-based schema.  Then, the workflow bundles the data into a ZIP archive.  The schema is owned by ClairCore.
+
+A Scanner instance runs the Vulnerability Updater (aka. Updater) to fetch the bundle and use ClairCore to parse the schema and update the vulnerability database.  Suppose ClairCore changed the schema in a backward-incompatible manner, and that change is incorporate in the Scanner repository (a ClairCore version bump).  In that scenario, the next Exporter execution will update the bundle with the new schema.  Then, existing Scanner instances won't be able to consume the data.  A Scanner version bump to a version of ClairCore supporting the new schema would be required.
+
+To address that scenario, Scanner created one bundle per Stackrox release.  Each bundle is created with an Exporter running the same ClairCore version used to build the release, ensuring compatibility.  Bundles are served at `https://definitions.stackrox.io/v4/V/vulnerabilities.zip` where `V` will be a Stackrox release version such as "4.5.0".  Scanner uses its release version tagged during build time to determine which version to fetch from.  The versions to build bundles for are taken from a file named `scanner/updater/version/RELEASE_VERSION`.
+
+For its purpose, this approach needs to be revised.
+
+1. The vulnerability schema rarely changes, but maintaining multiple bundles is expensive.  It requires multiple GitHub workflow runs, which increases the chances of failure and pressure on the security sources' endpoints.
+2. It isn't easy to backport fixes and updates to previous releases.  This requires patch releases, which trigger the release workflow.  Improving the release process is noisy, convoluted, and complex.
+3. The offline mode's "latest bundle" needs to exist as a complex solution to help users transition to the versioned bundle.  It contains all versions in a single bundle.  But one bundle per release makes the latest bundle large and, most of the time, essentially carries the same data since the schema is the same.
```

Comment:
The offline bundle is versioned as well, `roxctl scanner download-db` was created to address this concern to make it easier for users.  The commitment was for the latest archive to include the `4.4[.x]` offline bundles, no commitment was made to include `4.5[.x]` in the bundles.  The [docs](https://docs.openshift.com/acs/4.4/configuration/enable-offline-mode.html#download-scanner-definitions_enable-offline-mode) were also updated to recommend using `roxctl scanner download-db` over the latest archive.